js/apps/admin-ui/src/events/UserEvents.tsx

```diff
@@ -92,23 +95,26 @@ const DetailCell = (event: EventRepresentation) => (
 const UserDetailLink = (event: EventRepresentation) => {
   const { t } = useTranslation();
   const { realm } = useRealm();
+  // Extract the username from event.details, if it available
+  const userName = event.details ? event.details["username"] : null;
+  const userId = event.userId;
 
-  return (
-    <>
-      {event.userId && (
-        <Link
-          key={`link-${event.time}-${event.type}`}
-          to={toUser({
-            realm,
-            id: event.userId,
-            tab: "settings",
-          })}
-        >
-          {event.userId}
-        </Link>
-      )}
-      {!event.userId && t("noUserDetails")}
-    </>
+  return userId && userName ? (
```

Comment:
This change now requires both userId and userName to be set. But there are a lot of old events in the database that have a userId, but no username. And those are connected to a user. Those should be displayed with the userID and should link to the user (as before).